src/ed25519.rs

```diff
@@ -562,34 +563,30 @@ impl ExpandedSecretKey {
     }
 
     /// Sign a message with this `ExpandedSecretKey`.
+    #[allow(non_snake_case)]
     pub fn sign<D>(&self, message: &[u8], public_key: &PublicKey) -> Signature
             where D: Digest<OutputSize = U64> + Default {
         let mut h: D = D::default();
-        let mut hash: [u8; 64] = [0u8; 64];
-        let mesg_digest: Scalar;
-        let hram_digest: Scalar;
-        let r: CompressedEdwardsY;
+        let R: CompressedEdwardsY;
+        let r: Scalar;
         let s: Scalar;
+        let k: Scalar;
 
         h.input(&self.nonce);
         h.input(&message);
-        hash.copy_from_slice(h.fixed_result().as_slice());
-
-        mesg_digest = Scalar::from_bytes_mod_order_wide(&hash);
 
-        r = (&mesg_digest * &constants::ED25519_BASEPOINT_TABLE).compress();
+        r = Scalar::from_hash(h);
+        R = (&r * &constants::ED25519_BASEPOINT_TABLE).compress();
```

Comment:
I think it would be cleaner to do `let R = ...` here instead of defining the variable earlier